packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -476,35 +441,134 @@ export function createRouteManifest(
 			generate,
 			pathname: pathname || void 0,
 			prerender: false,
-			redirect: to,
-			redirectRoute: routes.find((r) => r.route === to),
+			redirect: 'status' in to
+				? {
+					status: to.status,
+					destination: to.destination,
+				}
+				: to.destination,
+			redirectRoute: routeMap.get(to.destination),
 			fallbackRoutes: [],
-		};
+		});
+	});
 
-		const lastSegmentIsDynamic = (r: RouteData) => !!r.segments.at(-1)?.at(-1)?.dynamic;
+	return routes;
+}
 
-		const redirBase = path.posix.dirname(route);
-		const dynamicRedir = lastSegmentIsDynamic(routeData);
-		let i = 0;
-		for (const existingRoute of routes) {
-			// An exact match, prefer the page/endpoint. This matches hosts.
-			if (existingRoute.route === route) {
-				routes.splice(i + 1, 0, routeData);
-				return;
+function routesCollide(a: RouteData, b: RouteData) {
+	if (a.segments.length !== b.segments.length) {
+		return false;
+	}
+
+	if (a.type === 'fallback' || b.type === 'fallback') {
+		// If either route is a fallback route, they don't collide.
+		// Fallbacks are always added below other routes exactly to avoid collisions.
+		return false;
+	}
+
+	const length = a.segments.length;
```

Comment:
Why do we choose the lengh of `a` and not `b`? Now I question why we call the two parameters `a` and `b`. Shouldn't we choose the great lenght between `a` and `b`? Otherwise if `b.length` is lower than `a.length`, we overflow.